### ISSUE TITLE
GAlert/GitAlert/QuitAlert: duplicated code

### ISSUE THREAD

**Issue by jackburton79** [Genio-The-Haiku-IDE/Genio#445]
GAlert, QuitAlert and GitAlert have a lot of code in common.
They are also use a lot of BAlert code.

We should try to do some refactor to reduce code duplication

**Comment by nexus6-haiku** [comment]
I agree, unfortunately BAlert can’t be extended so we may need to rewrite GAlert to be as much general as possible especially when it comes to managing buttons and validation rules.

**Comment by jackburton79** [comment]
Yeah that's the idea I guess.
